packages/app/cypress/e2e/runner/support/mochaEventsUtils.ts

```diff
@@ -0,0 +1,189 @@
+  lifecycle: () => 'match.number',
+  fnDuration: () => 'match.number',
+  duration: () => 'match.number',
+  afterFnDuration: () => 'match.number',
+  wallClockDuration: () => 'match.number',
+  stack: () => 'match.string',
+  file: (arg: string) => arg ? 'relative/path/to/spec.js' : undefined,
+  message: () => '[error message]',
+  sourceMappedStack: () => 'match.string',
+  parsedStack: () => 'match.array',
+  name: (n: string) => n === 'Error' ? 'AssertionError' : n,
+  err: () => {
+    return {
+      message: '[error message]',
+      name: 'AssertionError',
+      stack: 'match.string',
+      sourceMappedStack: 'match.string',
+      parsedStack: 'match.array',
+    }
+  },
+  startTime: new Date(0),
+  start: () => 'match.date',
+  end: () => 'match.date',
+  timings: (arg: MochaLifecycleData): SanitizedMochaLifecycleData => {
+    let sanitizedLifecycleData: SanitizedMochaLifecycleData = {
+      lifecycle: 'match.number',
+    }
+
+    for (const hook of hooks.before) {
+      const hooksToSanitize = arg[hook]
+
+      if (!hooksToSanitize) {
+        continue
+      }
+
+      sanitizedLifecycleData[hook] = hooksToSanitize.map((oldHook) => {
+        return {
+          hookId: oldHook.hookId,
+          fnDuration: 'match.number',
+          afterFnDuration: 'match.number',
+        }
+      })
+    }
+
+    if (arg.test) {
+      sanitizedLifecycleData.test = {
+        fnDuration: 'match.number',
+        afterFnDuration: 'match.number',
+      }
+    }
+
+    for (const hook of hooks.after) {
+      const hooksToSanitize = arg[hook]
+
+      if (!hooksToSanitize) {
+        continue
+      }
+
+      sanitizedLifecycleData[hook] = hooksToSanitize.map((oldHook) => {
+        return {
+          hookId: oldHook.hookId,
+          fnDuration: 'match.number',
+          afterFnDuration: 'match.number',
+        }
+      })
+    }
+
+    return sanitizedLifecycleData
+  },
+}
+
+const keysToEliminate = ['codeFrame', '_testConfig'] as const
+
+function removeUnusedKeysForTestSnapshot<T> (obj: T): T {
+  for (const key of keysToEliminate) {
+    delete obj[key]
+  }
+
+  for (const [key, value] of Object.entries(obj)) {
+    if (key in obj) {
+      const transform = eventCleanseMap[key]?.(value)
+
+      if (!transform) {
+        continue
+      }
+
+      obj[key] = transform
+    } else {
+      delete obj[key]
+    }
+  }
+
+  return obj
+}
+
+declare global {
+  interface Window {
+    bus: EventEmitter
+  }
+}
+
+class SnapshotError extends Error {
+  constructor (message: string) {
+    super()
+    this.message = `\n${message}`
+  }
+}
+
+export function runCypressInCypressMochaEventsTest<T> (snapshots: T, snapToCompare: keyof T, done: Mocha.Done) {
+  const bus = new EventEmitter()
+  const outerRunner = window.top!.window
+
+  outerRunner.bus = bus
+
+  // TODO: Can we automate writing the snapshots to disk?
+  // For some reason `cy.task('getSnapshot')` has problems when executed in
+  // "cypress in cypress"
```

Comment:
Let's log a ticket for this and reference here if we have something concrete to spike on. I don't think any other specs use that task at this point; when the old runner packages get removed, we will lose what little context we currently have around this.

Reply:
Great idea, done: https://cypress-io.atlassian.net/browse/UNIFY-1383. Let me know if you think this needs more details, is missing info, etc.

Reply:
Do we still need to add the ticket to the comment here?